src/realm/obj.hpp

```diff
@@ -117,17 +117,29 @@ class Obj {
     template <typename U>
     U get(ColKey col_key) const;
 
+    bool has_property(StringData prop_name) const;
+
+    std::vector<StringData> get_additional_properties() const;
```

Comment:
Can't there be a problem that we return references?

Reply:
No bigger problem than anywere else where we return pointers directly into the file.